drivers/mtd/ftl.c

```diff
@@ -172,6 +228,163 @@ static int ftl_close(FAR struct inode *inode)
   return OK;
 }
 
+/****************************************************************************
+ * Name: ftl_mtd_bread
+ *
+ * Description:
+ *   Read the specified number of sectors. If mtd device is nor flash, it
+ *   can be read once time. If mtd device is nand flash, it can be read one
+ *   block every time and need to skip bad block until the specified number
+ *   of sectors finish.
+ *
+ ****************************************************************************/
+
+static ssize_t ftl_mtd_bread(FAR struct ftl_struct_s *dev, off_t startblock,
+                             size_t nblocks, FAR uint8_t *buffer)
+{
+  off_t mask = dev->blkper - 1;
+  size_t nread = nblocks;
+  ssize_t ret = OK;
+
+  if (dev->lptable == NULL)
+    {
+      ret = MTD_BREAD(dev->mtd, startblock, nblocks, buffer);
+      if (ret != nblocks)
+        {
+          ferr("ERROR: Read %zu blocks starting at block %" PRIdOFF
+               " failed: %zd\n", nblocks, startblock, ret);
+        }
+
+      return ret;
+    }
+
+  while (nblocks > 0)
+    {
+      off_t startphysicalblock;
+      off_t starteraseblock;
+      size_t count;
+
+      starteraseblock = startblock / dev->blkper;
+      if (starteraseblock >= dev->lpcount)
+        {
+          ret = -ENOSPC;
+          break;
+        }
+
+      count = ftl_get_cblock(dev, starteraseblock, (nblocks + mask) & ~mask);
```

Comment:
@Donny9 should we divide nblocks by dev->blkper before pass to ftl_get_cblock?

Reply:
yes，The parameter count of ftl get_clock indicates the number of eraseblocks, but we have already done the conversion by `(nblocks + mask) & ~mask)`